lib/OpenQA/WebSockets/Controller/Worker.pm

```diff
@@ -172,62 +168,25 @@ sub _message {
                 # Tell the worker that we saw it (used for tests and debugging)
                 $tx->send({json => {type => 'info', seen => 1}});
             }
-            catch {
-                log_error("Failed updating seen and error status of worker $worker_id: $_");    # uncoverable statement
-            };
+            catch ($e) {
+                log_error("Failed updating seen and error status of worker $worker_id: $e");    # uncoverable statement
+            }
         }
 
         # find the job currently associated with that worker and check whether the worker still
         # executes the job it is supposed to
         my $current_job_id;
         try {
-            return undef unless $worker;
-
-            my $registered_job_token;
-            my $current_job_state;
-            my @unfinished_jobs = $worker->unfinished_jobs;
-            my $current_job = $worker->job // $unfinished_jobs[0];
-            if ($current_job) {
-                $current_job_id = $current_job->id;
-                $current_job_state = $current_job->state;
-            }
-
-            # log debugging info
-            log_trace("Found job $current_job_id in DB from worker_status update sent by worker $worker_id")
-              if defined $current_job_id;
-            log_trace("Received request has job id: $job_id")
-              if defined $job_id;
-            $registered_job_token = $worker->get_property('JOBTOKEN');
-            log_trace("Worker $worker_id for job $current_job_id has token $registered_job_token")
-              if defined $current_job_id && defined $registered_job_token;
-            log_trace("Received request has token: $job_token")
-              if defined $job_token;
-
-            # skip any further actions if worker just does the one job we expected it to do
-            return undef
-              if ( defined $job_id
-                && defined $current_job_id
-                && defined $job_token
-                && defined $registered_job_token
-                && $job_id eq $current_job_id
-                && (my $job_token_correct = $job_token eq $registered_job_token)
-                && OpenQA::Jobs::Constants::meta_state($current_job_state) eq OpenQA::Jobs::Constants::EXECUTION)
-              && (scalar @unfinished_jobs <= 1);
-
-            # give worker a second chance to process the job assignment
-            # possible situation on the worker: The worker might be sending a status update claiming it is
-            # idle (or has doing that task piled up on the event loop). At the same time a job arrives. The
-            # message regarding that job will be processed after sending the idle status. So let's give the
-            # worker another change to process the message about its assigned job.
-            return undef unless $worker_previously_idle;
-
+            my ($current_job, $unfinished_jobs)
```

Comment:
At the first glance that method is not returning those two values. Have to have a closer look

Reply:
true. I just rushed. moment. I will mark `not-ready` and look at it after meeting

Reply:
Previously the code would not call `reschedule_assigned_jobs` in certain cases. Now you call `_check_assignment_process` but you don't do anything depending on what the method returns, so `reschedule_assigned_jobs`  is always called.

Reply:
I think this is solved as part of https://github.com/os-autoinst/openQA/pull/6228#discussion_r1975244731. Can you check again?